theano/tensor/nnet/tests/test_abstract_conv.py

```diff
@@ -781,3 +783,77 @@ def test_compare_1D_and_2D_upsampling_values(self):
         f_1D = theano.function([], mat_1D, mode=self.compile_mode)
         f_2D = theano.function([], mat_2D, mode=self.compile_mode)
         utt.assert_allclose(f_1D(), f_2D(), rtol=1e-06)
+
+
+def test_conv2d_grad_wrt_inputs():
+
+    inputs_shapes = [(8, 1, 12, 12), (8, 1, 18, 18), (2, 1, 4, 4),
+                     (6, 1, 10, 11), (2, 1, 6, 5), (1, 5, 9, 9)]
+    filters_shapes = [(5, 1, 2, 2), (4, 1, 3, 3), (2, 1, 3, 3),
+                      (1, 1, 2, 5), (4, 1, 2, 2), (4, 5, 2, 2)]
+    subsamples = [(1, 1), (2, 2)]
+    border_modes = ["valid", "full"]
+    filter_flip = [True, False]
+    x = T.cast(T.ftensor4('x'), theano.config.floatX)
+    for (in_shape, fltr_shape) in zip(inputs_shapes, filters_shapes):
+        input_val = np.random.rand(*in_shape).astype(theano.config.floatX)
+
+        filter_val = np.random.rand(*fltr_shape).astype(theano.config.floatX)
+        for bm in border_modes:
+            for ss in subsamples:
+                for ff in filter_flip:
+                    cgi = conv2d_grad_wrt_inputs(output_grad=input_val,
+                                                 filters=filter_val,
+                                                 filter_shape=fltr_shape,
+                                                 input_shape=in_shape,
+                                                 border_mode=bm,
+                                                 subsample=ss,
+                                                 filter_flip=ff)
```

Comment:
It would be better to use symbolic (or shared) variables for `input` (or `output_grad`) and `filters`, so constant-folding does not get in the way during optimization.